eppo_client/client.py

```diff
@@ -183,114 +187,63 @@ def get_assignment_variation(
         """
         validate_not_blank("subject_key", subject_key)
         validate_not_blank("flag_key", flag_key)
-        experiment_config = self.__config_requestor.get_configuration(flag_key)
-        override = self._get_subject_variation_override(experiment_config, subject_key)
-        if override:
-            if expected_variation_type is not None:
-                variation_is_expected_type = VariationType.is_expected_type(
-                    override, expected_variation_type
-                )
-                if not variation_is_expected_type:
-                    return None
-            return override
+        if subject_attributes is None:
+            subject_attributes = {}
```

Comment:
why?

Reply:
So that from that point onwards we can assume subject_attributes is a dictionary and can avoid the "null" case